### PR TITLE
entry draft for a possible new header for platform compiler support

### PR DESCRIPTION
**Pull request by Emmankoko**
This PR is to propose an addition for Platform compiler support header to `cpp_interface.dd` file being the interfacing to C++ reference page. Might be verbose but I think it's needed especially for beginners. I remember nearly giving up when I started fresh with C++ interop because I was running my C++ code with g++ on windows and being linked to my D code compiled with DMD. It was a next level headache being a beginner.
so it can just be a simple instruction to let users know that the following documentation works with this compiler1-compiler2 pair on platform A. Ditto for platform B and platform C.

@Geod24

### CODE REVIEW

**Comment by dkorpel** [comment]
What's the status of this, are you waiting for approval to write the content? It's hard to judge whether it fits in this spot when there's only a header text, but whatever you end up writing, I'm sure we can find a place for it; be it here, in an article, or on a wiki page.

**Comment by Emmankoko** [comment]
opened for review.....

File: spec/cpp_interface.dd
Comment (dkorpel) [inline spec/cpp_interface.dd:48]:
```suggestion
    You can link to both gcc and clang compiled binaries on linux.
```

**Review by dkorpel — APPROVED** [review]

**Pull request merged**
